lib/piezo.js

```diff
@@ -129,202 +137,284 @@ Piezo.Notes = {
     }
   }
-  return song;
+
+  return frequency;
 }
 
-Piezo.prototype.play = function(tune, callback) {
-  if (typeof tune !== "object") {
-    tune = {
-      song: tune
-    };
+Piezo.prototype.pulsePin = function(pulse, duration) {
+  duration = duration || this.defaults.duration;
+
+  if (isNaN(pulse) || isNaN(duration)) {
+    this.clearTone();
   }
-  if (typeof tune.song === "string") {
-    tune.song = stringToSong(tune.song, tune.beats);
+
+  clearTimer.call(this);
+  var durationTimer = this.timer = new Timer();
+
+  var pinState = 1;
+  var onPulse = function() {
+    pinState = pinState === 1 ? 0 : 1;
+    this.io.digitalWrite(this.pin, pinState);
+
+    var timeRunning = durationTimer.difTime / 1000000;
+    if (timeRunning > duration) {
+      clearTimer.call(this);
+    }
+  }.bind(this);
+
+  var pulseInterval = pulse + "u";
+  durationTimer.setInterval(onPulse, null, pulseInterval);
+
+  return this;
+};
+
+Piezo.prototype.playFrequency = function(frequency, duration) {
+  duration = duration || this.defaults.duration;
+
+  if (isNaN(frequency) || isNaN(duration)) {
+    this.clearTone();
   }
-  var duration,
-    tempo = tune.tempo || 250,
-    beatDuration = Math.round(60000 / tempo), // Length for a single beat in ms
-    song = tune.song || [],
-    i = 0;
-  if (song && !Array.isArray(song)) {
-    song = [song];
+
+  var pulse = frequencyToPulse(frequency);
+
+  return this.pulsePin(pulse, duration);
+};
+
+Piezo.prototype.playNote = function(note, duration) {
+  note = normalizeNote(note);
+  duration = duration || this.defaults.duration;
+
+  if (!note || isNaN(duration)) {
+    this.clearTone();
   }
-  var next = function() {
-    var note, beat;
 
-    note = parseTone(song[i]);
-    beat = parseBeat(song[i]);
+  var pulse = noteToPulse(note);
+
+  return this.pulsePin(pulse, duration);
+};
+
+Piezo.prototype.playMelody = function(melody, callback) {
+  melody = normalizeMelody(melody);
+
+  if (!melody) {
+    return callback(new Error("Piezo.playMelody invalid melody"));
+  }
 
-    duration = beat * beatDuration;
+  var index = 0;
+  var playNextNote = function() {
+    var note = melody.notes[index];
 
-    if (i++ === song.length) {
-      // Song is over
+    if (index++ === melody.notes.length) {
       this.isPlaying = false;
-      if (typeof callback === "function") {
-        callback(tune);
-      }
-      return;
-    }
+      callback(null);
 
-    if (note === null) {
-      this.noTone();
-    } else {
-      this.frequency(note, duration);
+      return this;
     }
 
-    setTimeout(next, duration);
+    var duration = note.beat * melody.beatDuration;
+    this.playNote(note, duration);
+
+    setTimeout(playNextNote, duration);
   }.bind(this);
 
-  // We are playing a song
   this.isPlaying = true;
 
-  next();
+  playNextNote();
 
   return this;
 };
 
-Piezo.prototype.noTone = function() {
+Piezo.prototype.stop = function() {
   this.io.digitalWrite(this.pin, 0);
   clearTimer.call(this);
```

Comment:
Let's change this to match `Led`: 

- `stop` just stops an internal timer, but doesn't change the state 
- `off` changes the state to off

It's a slight change from before, but we should be ok with breaking small things in the name of consistency.